### PR TITLE
Merge 2.3 to master and fix coding standards

### PR DESCRIPTION
**Pull request by hypeJunction**

### CODE REVIEW

**Comment by hypeJunction** [comment]
I think I got it right this time. The only downside is we will probably have major conflicts when merging up from 2.3, perhaps we should limit the amount of work we do there anyway.

File: actions/admin/plugins/activate.php [hunk 1]
```diff
@@ -50,17 +49,14 @@
...
 	$url = 'admin/plugins';
-	$query = (string)parse_url($_SERVER['HTTP_REFERER'], PHP_URL_QUERY);
+	$query = (string) parse_url($_SERVER['HTTP_REFERER'], PHP_URL_QUERY);
 	if ($query) {
 		$url .= "?$query";
 	}
 	$plugin = get_entity($plugin_guids[0]);
 	$id = $css_id = preg_replace('/[^a-z0-9-]/i', '-', $plugin->getID());
 	$url = "$url#id";
-	$data = [
```
Comment (jdalsem) [inline actions/admin/plugins/activate.php]: is this removal intended

**Comment by jdalsem** [comment]
> perhaps we should limit the amount of work we do there anyway

it already is bugfix only, so i think that won't be that much

File: actions/admin/plugins/activate.php (hunk 1 above)
Comment (hypeJunction) [inline actions/admin/plugins/activate.php]: Must be from 2.3 merge

Comment (hypeJunction) [inline actions/admin/plugins/activate.php]: Yes, it's something that @mrclay fixed in 2.3

**Comment by hypeJunction** [comment]
So, in summary:
- `phpcs` rules are similar to what we've had, maybe a little more relaxed (line length is PSR2 so not so strict; some warning rules about line endings etc, which are not treated as errors)
- `phpcbf` fixes everything it can whenever it's ran. We can do ``composer lint-fixer`` not and then to fix minor standard violations

File: engine/classes/ElggEntity.php [hunk 2]
```diff
@@ -1602,13 +1602,13 @@ public function disable($reason = "", $recursive = true) {
 			$ia = elgg_set_ignore_access(true);
 
 			$base_options = [
-				'wheres' => [
+					'wheres' => [
 					"e.guid != $guid",
```
Comment (jdalsem) [inline engine/classes/ElggEntity.php]: incorrect indent

File: engine/classes/ElggEntity.php [hunk 3]
```diff
@@ -1768,25 +1768,25 @@ public function delete($recursive = true) {
...
 			$ia = elgg_set_ignore_access(true);
-
+			
 			// @todo there was logic in the original code that ignored
 			// entities with owner or container guids of themselves.
 			// this should probably be prevented in \ElggEntity instead of checked for here
 			$base_options = [
-				'wheres' => [
+					'wheres' => [
 					"e.guid != $guid",
```
Comment (jdalsem) [inline engine/classes/ElggEntity.php]: incorrect indent

File: engine/classes/ElggEntity.php (hunk 2 above)
Comment (hypeJunction) [inline engine/classes/ElggEntity.php]: I don't care now. Just want this monster out of my way.

Comment (jdalsem) [inline engine/classes/ElggEntity.php]: but why is phpcs not complaining about this?

Comment (hypeJunction) [inline engine/classes/ElggEntity.php]: It never complains about array element indentation

Comment (hypeJunction) [inline engine/classes/ElggEntity.php]: https://github.com/squizlabs/PHP_CodeSniffer/issues/1235

**Comment by hypeJunction** [comment]
All green. @mrclay do you mind taking a look and pressing the green button? The merge part is yours, I added the cs part

File: engine/classes/ElggEntity.php (hunk 2 above)
Comment (jdalsem) [inline engine/classes/ElggEntity.php]: 👍

**Comment by jdalsem** [comment]
why are there commits in the PR that are already in master?

**Comment by hypeJunction** [comment]
Such as? @mrclay did the merge PR yesterday

**Comment by jdalsem** [comment]
> why are there commits in the PR that are already in master?

nevermind... was confused by all the 2.3.0 commits

**Comment by mrclay** [comment]
Before I saw this I started work on the minimal PR to get the phpcs working #10831. This seems like a better idea: get 2.3 builds working, then merge into master, then apply rest of this on master in a separate PR.

**Comment by hypeJunction** [comment]
I can live with that, but someone needs to redo the pull requests. I am spending my holiday working on fixing Elgg sniffs. How decadent of me.

**Comment by hypeJunction** [comment]
@jdalsem Now is your chance to move sniffs to the repo.
